pkg/security/secl/rules/policy_test.go

```diff
@@ -432,14 +434,75 @@ func loadPolicy(t *testing.T, testPolicy *PolicyDef) *multierror.Error {
 		t.Fatal(err)
 	}
 
-	provider, err := NewPoliciesDirProvider(tmpDir, false)
+	provider, err := NewPoliciesDirProvider(tmpDir, false, agentVersion)
 	if err != nil {
 		t.Fatal(err)
 	}
 
 	loader := NewPolicyLoader(provider)
 
-	return rs.LoadPolicies(loader)
+	return rs, rs.LoadPolicies(loader)
+}
+
+func TestRuleAgentConstraint(t *testing.T) {
+
+	testEntries := []struct {
+		name           string
+		agentVersion   string
+		ruleConstraint string
+		expectLoad     bool
+	}{
+		{
+			name:           "basic",
+			agentVersion:   "7.38",
+			ruleConstraint: "< 7.37",
+			expectLoad:     false,
+		},
+		{
+			name:           "basic2",
+			agentVersion:   "7.35",
+			ruleConstraint: "< 7.37",
+			expectLoad:     true,
+		},
+		{
+			name:           "range",
+			agentVersion:   "7.35",
+			ruleConstraint: ">= 7.30, < 7.37",
+			expectLoad:     true,
+		},
+		{
+			name:           "range_not",
+			agentVersion:   "7.35",
+			ruleConstraint: ">= 7.30, < 7.37, != 7.35",
+			expectLoad:     false,
+		},
+	}
+
+	for _, entry := range testEntries {
+		t.Run(entry.name, func(t *testing.T) {
+			ruleID := fmt.Sprintf("test_rule_%s", entry.name)
+
+			testPolicy := &PolicyDef{
+				Rules: []*RuleDefinition{{
+					ID:                     ruleID,
+					Expression:             `open.filename == "/tmp/test"`,
+					AgentVersionConstraint: entry.ruleConstraint,
+				}},
+			}
+
+			agentVersion, err := semver.NewVersion(entry.agentVersion)
+			assert.Nil(t, err)
+
+			rs, err := loadPolicy(t, testPolicy, agentVersion)
+			assert.Nil(t, err)
+
+			if entry.expectLoad {
+				assert.Contains(t, rs.rules, ruleID)
+			} else {
+				assert.NotContains(t, rs.rules, ruleID)
+			}
+		})
+	}
 }
```

Comment:
Could we add a test to make sure that we properly ignore rules that failed to parse (if we don't have such test yet) ?

Reply:
Good idea, I will add one

Reply:
Does the last commit work for you ?